File: 1477-product-of-the-last-k-numbers/1477-product-of-the-last-k-numbers.java
class ProductOfNumbers {
    private List<Integer> l;
    public ProductOfNumbers() {
        l = new ArrayList<>();
        l.add(1);
    }
    
    public void add(int num) {
        if(num == 0){
            l = new ArrayList<>();
            l.add(1);
        }else{
            int new_num = l.get(l.size()-1);
            l.add(new_num*num);
        }
    }
    
    public int getProduct(int k) {
        if(k >= l.size()){
            return 0;
        }
        return l.get(l.size()-1)/l.get(l.size()-1-k);
    }
}

/**
 * Your ProductOfNumbers object will be instantiated and called as such:
 * ProductOfNumbers obj = new ProductOfNumbers();
 * obj.add(num);
 * int param_2 = obj.getProduct(k);
 */

File: 1477-product-of-the-last-k-numbers/1477-product-of-the-last-k-numbers.py
class ProductOfNumbers:

    def __init__(self):
        self.l = [1]

    def add(self, num: int) -> None:
        if num == 0:
            self.l = [1]
        else:
            new_num = self.l[-1]
            self.l.append(new_num*num)

    def getProduct(self, k: int) -> int:
        if len(self.l) <= k:
            return 0
        
        return self.l[-1] // self.l[-1-k]


# Your ProductOfNumbers object will be instantiated and called as such:
# obj = ProductOfNumbers()
# obj.add(num)
# param_2 = obj.getProduct(k)